httpcore5/src/main/java/org/apache/hc/core5/pool/StrictConnPool.java

```diff
@@ -171,21 +171,33 @@ public void close() {
         Args.notNull(route, "Route");
         Args.notNull(requestTimeout, "Request timeout");
         Asserts.check(!this.isShutDown.get(), "Connection pool shut down");
+        final Deadline deadline = Deadline.calculate(requestTimeout);
         final BasicFuture<PoolEntry<T, C>> future = new BasicFuture<>(callback);
-        this.lock.lock();
+        boolean acquiredLock = false;
+
         try {
-            final LeaseRequest<T, C> request = new LeaseRequest<>(route, state, requestTimeout, future);
-            final boolean completed = processPendingRequest(request);
-            if (!request.isDone() && !completed) {
-                this.leasingRequests.add(request);
-            }
-            if (request.isDone()) {
-                this.completedRequests.add(request);
+            acquiredLock = this.lock.tryLock(requestTimeout.getDuration(), requestTimeout.getTimeUnit());
+        } catch (final InterruptedException ignored) {
```

Comment:
@cwildman `InterruptedException` handling logic does not look quite right to me. At the very least one must call                 `Thread.currentThread().interrupt()` when catching and not re-throwing `InterruptedException`. I also think we should either cancel request at that point or propagate the exception to the caller.

Reply:
Makes sense. I updated to set interrupted again and immediately return a failed future with the InterruptedException.